astropy/stats/lombscargle/implementations/mle.py

```diff
@@ -0,0 +1,98 @@
+"""Tools for maximum likelihood estimation associated with Lomb-Scargle"""
+import numpy as np
+
+
+def design_matrix(t, frequency, dy=None, bias=True, nterms=1):
+    """Compute the Lomb-Scargle design matrix at the given frequency
+
+    This is the matrix X such that the periodic model at the given frequency
+    can be expressed :math:`\\hat{y} = X \\theta`.
+
+    Parameters
+    ----------
+    t : array_like, shape=(n_times,)
+        times at which to compute the design matrix
+    frequency : float
+        frequency for the design matrix
+    dy : float or array_like (optional)
+        data uncertainties: should be broadcastable with `t`
+    bias : bool (default=True)
+        If true, include a bias column in the matrix
+    nterms : int (default=1)
+        Number of Fourier terms to include in the model
+
+    Returns
+    -------
+    X : ndarray, shape=(n_times, n_parameters)
+        The design matrix, where n_parameters = bool(bias) + 2 * nterms
+    """
+    t = np.asarray(t)
```

Comment:
Like above, I think one should be able to *assume* input is array-like, so would remove it.  If you prefer to have it, do use `np.asanyarray`.